tests/phpunit/tests/interactivity-api/wpInteractivityAPI.php

```diff
@@ -367,6 +367,249 @@ public function test_state_and_config_escape_special_characters() {
 		);
 	}
 
+	/**
+	 * Test that calling state without a namespace arg returns the state data
+	 * for the current namespace in the internal namespace stack.
+	 *
+	 * @ticket 61037
+	 *
+	 * @covers ::state
+	 */
+	public function test_state_without_namespace() {
+		$interactivity = new ReflectionClass( $this->interactivity );
```

Comment:
Would it be possible to rewrite the tests to avoid using reflections and exposing all the internals? It's very difficult to say how it all works just by looking at the test. One of the benefits of having code coverage is also documenting how the API can be used.

Reply:
I agree, this is difficult to follow and seems to rely a lot on internals.

This sounds like a big refactor (maybe it could be pulled out to its own PR landed before this) but could we do something where we reset the global Interactivity API between tests in setup or teardown methods? _Maybe_ that could use Reflection if absolutely necessary, but at least it should be isolate to one place.

I'd really like to see tests setting declaring their own state. I think the tests would become easier to read for the most part and it would make it easier to understand when there are multiple namespaces in play.

Reply:
I'm OK leaving this for now.

Reply:
> I'd really like to see tests setting declaring their own state. I think the tests would become easier to read for the most part and it would make it easier to understand when there are multiple namespaces in play.

@gziolo, @sirreal, I've created two helper methods to define the internal namespace and context stacks and moved the relevant state to each test case. At least the test code is easier to understand and maintain this way.

Reply:
That seems much better, thanks.